manager_dashboard/manager_dashboard/js/utils.js

```diff
@@ -0,0 +1,134 @@
+const getFormData = object => Object.keys(object).reduce((formData, key) => {
+    formData.append(key, object[key]);
+    return formData;
+}, new FormData());
+
+
+function geojsonToFeatureCollection(geojson){
+    if (geojson.type != "FeatureCollection"){
+        var collection = {
+            type: 'FeatureCollection',
+            features: [{"type":"feature", "geometry":geojson}]
+        };
+        return collection;
+    }
+    return geojson;
+}
+
+
+async function projectAoi(TMId){
+    let response = await fetch(`https://tasking-manager-tm4-production-api.hotosm.org/api/v2/projects/${TMId}/queries/aoi/?as_file=false`);
+    const answer = await response.json();
+    return answer
+}
+
+
+async function ohsome_request(url, data){
+    let response = await fetch(url, {
+        "method": 'POST',
+        "body": data,
+    })
+    const answer_ohsome = await response.json();
+    return answer_ohsome
+}
+
+
+async function countObjectsInFilter(bpolys, filter){
+    let url = "https://api.ohsome.org/v1/elements/count"
+    let data = {"bpolys": JSON.stringify(geojsonToFeatureCollection(bpolys)), "filter": filter}
+    let answer = await ohsome_request(url, getFormData(data))
+
+    let count = answer.result[0].value
+    if ((count != null) & (count > 0) & (count < 100000)){
+        // valid result
+        return answer;
+    }
+    else {
+        if ((count > 100000)){
+            answer.Error = `Area of Interest contains more than 100 000 objects. -> ${count}`
+        }
+        else{
+            answer.Error = "Area of Interest does not contain objects from filter."
+        }
+        return answer
+    }
+}
+
+
+async function validateTMIdAndFilter(TMId, filter){
+    let answer_aoi = await projectAoi(TMId)
+    if (answer_aoi.Error != null){
+        return answer_aoi;
+    }
+    let answer = countObjectsInFilter(answer_aoi, filter)
+    if (answer.Error != null){
+        return answer;
+    }
+    return answer_aoi
+}
+
+
+async function validateTMId(event){
+    var infoOutput = $(event).parent().siblings(".inputInfo")[0];
+
+    TMId = $("#inputTaskGeometriesId").val()
+    if (isNaN(TMId) | TMId === ""){
+        infoOutput.innerHTML = "Input is not a Number"
+        return null
+    }
+    else{
+        answer = await projectAoi(TMId)
+        if (answer.Error != null){
+            infoOutput.innerHTML = "Project ID not found in HOT Tasking Manager"
+            return null
+        }
+        else {
+            infoOutput.innerHTML = "Valid TMId"
+            return answer
+        }
+    }
+}
+
+
+async function validateFilter(event){
```

Comment:
I think that this is not working as expected.

When I use TM id `3484` I am still able to upload the project to firebase. But I think this function should catch the we can't request the ohsome api for an invalid AOI.

Reply:
Missing await and looking for answer.Error instead of answer.status!=200 in the ohsome request led to this behavior. Should be fixed now :)